### PR TITLE
Php3branch > tl1revbranch

### PR DESCRIPTION
**Pull request by php-3**

### CODE REVIEW

File: portfolio/ifintech/v1/ierp/ierp-7.0.0/htdocs/custom/iexport/customagents/auditlog.php [hunk 1]
```diff
@@ -96,7 +96,71 @@
 //Log page goes here
 print $langs->trans("Custom Agent Status Audit Log");
 
-
+        
+/*****************************************************************************/
+/* view tabel                                                                */
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/ierp-7.0.0/htdocs/custom/iexport/customagents/auditlog.php]: table!

File: portfolio/ifintech/v1/ierp/ierp-7.0.0/htdocs/custom/iexport/customagents/card.php [hunk 2]
```diff
@@ -160,15 +160,22 @@
                 $customagentsobject->description = $description;
                 $customagentsobject->supplier_id = $supplier_id;
                 $customagentsobject->status=0;
-
+                $oldvalues = array();
+                $newvalues = array();
+                 
+                $oldvalues[0] = $customagentsobject->reference;
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/ierp-7.0.0/htdocs/custom/iexport/customagents/card.php]: better by name not by index

File: portfolio/ifintech/v1/ierp/ierp-7.0.0/htdocs/custom/iexport/customagents/card.php [hunk 3]
```diff
@@ -160,15 +160,22 @@
...
-
+                $newvalues = $oldvalues;
+                
                 $customagentsobject->user_id      = $fuserid ;
                 $customagentsobject->customagent_id  = $customagentsobject->id;
                 $customagentsobject->prevstatus   = 0;
 
                 $resultlog=$customagentsobject->saveStatusLog($user);
-
+                echo $customagentsobject->id.'//////////';
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/ierp-7.0.0/htdocs/custom/iexport/customagents/card.php]: remove this echo

File: portfolio/ifintech/v1/ierp/ierp-7.0.0/htdocs/custom/iexport/sql/couriers/llx_iexport_courier_auditlog.sql [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+ */
+/**
+ * Author:  php3
+ * Created: Mar 13, 2018
+ */
+DROP TABLE llx_iexport_courier_auditlog;
+CREATE TABLE llx_iexport_courier_auditlog (
+  rowid INTEGER(11) NOT NULL AUTO_INCREMENT PRIMARY KEY,
+  userid INTEGER(11) NOT NULL,
+  changedate DATETIME NOT NULL,
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/ierp-7.0.0/htdocs/custom/iexport/sql/couriers/llx_iexport_courier_auditlog.sql]: where is the field name ?

File: portfolio/ifintech/v1/ierp/ierp-7.0.0/htdocs/custom/iexport/sql/couriers/llx_iexport_courier_auditlog.sql [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+  courier_id INTEGER(11) NOT NULL,
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/ierp-7.0.0/htdocs/custom/iexport/sql/couriers/llx_iexport_courier_auditlog.sql]: make it "entity_id" only
and make it same for all tables

File: portfolio/ifintech/v1/ierp/ierp-7.0.0/htdocs/main.inc.php [hunk 6]
```diff
@@ -1050,6 +1050,126 @@ function llxHeader($head='', $title='', $help_url='', $target='', $disablejs=0,
  *  @param	int		$forcenocache	Force disabling of cache for the page
  *  @return	void
  */
+function fnCreateAuditLog($object , $oldvalues , $newvalues , $id)
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/ierp-7.0.0/htdocs/main.inc.php]: make name as dolibarr standard at this file not fnCreateAuditLog

File: portfolio/ifintech/v1/ierp/ierp-7.0.0/htdocs/main.inc.php [hunk 7]
```diff
@@ -1050,6 +1050,126 @@ function llxHeader($head='', $title='', $help_url='', $target='', $disablejs=0,
...
+   {
+        $sql = "INSERT INTO ".MAIN_DB_PREFIX.$object->table_element."_auditlog(";
+        $sql.= "userid,";
+        $sql.= "changedate,";
+        $sql.= "$id,";
+        $sql.= "oldvalue,";
+        $sql.= "newvalue";
+        $sql.= ") VALUES (";
+        $sql.= " $object->userid, ";
+        $sql.= " '".$object->changedate."' ,";
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/ierp-7.0.0/htdocs/main.inc.php]: why you added " '". at beginning of each row, it should be at the end of the previous row

File: portfolio/ifintech/v1/ierp/ierp-7.0.0/htdocs/main.inc.php [hunk 8]
```diff
@@ -1050,6 +1050,126 @@ function llxHeader($head='', $title='', $help_url='', $target='', $disablejs=0,
...
+        $sql.= " '".$object->id."', ";
+        $sql.= " '".$oldvalues[$i]."', ";
+        $sql.= " '".$newvalues[$i]."' ";
+        $sql.= ")";
+        echo $sql;
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/ierp-7.0.0/htdocs/main.inc.php]: remove this echo

File: portfolio/ifintech/v1/ierp/ierp-7.0.0/htdocs/main.inc.php [hunk 9]
```diff
@@ -1050,6 +1050,126 @@ function llxHeader($head='', $title='', $help_url='', $target='', $disablejs=0,
...
+        $resql=$object->db->query($sql);
+        if (! $resql) {
+            $error++; $object->errors[]="Error ".$object->db->lasterror();
+        }
+
+        if (! $error)
+        {
+            $object->auditlog_id = $object->db->last_insert_id(MAIN_DB_PREFIX.$object->table_element."_auditlog");
+
+            if (! $notrigger)
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/ierp-7.0.0/htdocs/main.inc.php]: where is this parameter come from ?? $notrigger

File: portfolio/ifintech/v1/ierp/ierp-7.0.0/htdocs/main.inc.php [hunk 10]
```diff
@@ -1050,6 +1050,126 @@ function llxHeader($head='', $title='', $help_url='', $target='', $disablejs=0,
...
+            $object->db->commit();
+        }
+   }
+   if ($error)
+        {
+            return -1*$error;
+        }
+        else
+        {
+            $object->db->commit();
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/ierp-7.0.0/htdocs/main.inc.php]: why 2 different places are doing commit??

File: portfolio/ifintech/v1/ierp/ierp-7.0.0/htdocs/main.inc.php [hunk 11]
```diff
@@ -1050,6 +1050,126 @@ function llxHeader($head='', $title='', $help_url='', $target='', $disablejs=0,
...
+            return $object->auditlog_id;
+        }
+}
+function fnFetchAuditLog ( $id , $object)
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/ierp-7.0.0/htdocs/main.inc.php]: fnFetchAuditLog is not as per dolibarr standard for this file
don't use our standard for custom modules here

Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/ierp-7.0.0/htdocs/main.inc.php]: $id no need for this parameter if the field name is same at the tables.

File: portfolio/ifintech/v1/ierp/ierp-7.0.0/htdocs/custom/iexport/customagents/card.php (hunk 2 above)
Comment (php-3) [reply portfolio/ifintech/v1/ierp/ierp-7.0.0/htdocs/custom/iexport/customagents/card.php]: ok

File: portfolio/ifintech/v1/ierp/ierp-7.0.0/htdocs/custom/iexport/customagents/card.php (hunk 3 above)
Comment (php-3) [reply portfolio/ifintech/v1/ierp/ierp-7.0.0/htdocs/custom/iexport/customagents/card.php]: done

File: portfolio/ifintech/v1/ierp/ierp-7.0.0/htdocs/custom/iexport/sql/couriers/llx_iexport_courier_auditlog.sql (hunk 4 above)
Comment (php-3) [reply portfolio/ifintech/v1/ierp/ierp-7.0.0/htdocs/custom/iexport/sql/couriers/llx_iexport_courier_auditlog.sql]: i create it in last pull request

File: portfolio/ifintech/v1/ierp/ierp-7.0.0/htdocs/custom/iexport/sql/couriers/llx_iexport_courier_auditlog.sql (hunk 5 above)
Comment (php-3) [reply portfolio/ifintech/v1/ierp/ierp-7.0.0/htdocs/custom/iexport/sql/couriers/llx_iexport_courier_auditlog.sql]: ok

**Pull request merged**